docs/INSTALL.K8S.md

```diff
@@ -4,26 +4,85 @@ Installing Kubernetes
 On the master node, start etcd
 
 ```
-docker run --net=host -d gcr.io/google_containers/etcd:2.0.12 /usr/local/bin/etcd --addr=127.0.0.1:4001 --bind-addr=0.0.0.0:4001 --data-dir=/var/etcd/data
+docker run \
+  --net=host \
+  --detach \
+  gcr.io/google_containers/etcd:2.0.12 \
+  /usr/local/bin/etcd \
+    --addr=127.0.0.1:4001 \
+    --bind-addr=0.0.0.0:4001 \
+    --data-dir=/var/etcd/data
 ```
 
-Download the latest stable kubernetes.tar.gz from: https://github.com/kubernetes/kubernetes/releases
+Download the latest stable kubernetes.tar.gz from:
+https://github.com/kubernetes/kubernetes/releases
 
-Untar the file and look for kubernetes-server-linux-amd64.tar.gz.
-Untar that file too.  Copy kube-apiserver, kube-controller-manager,
-kube-scheduler, kubelet and kubectl to a directory.
+Untar the file and look for kubernetes-server-linux-amd64.tar.gz. Untar that
+file too. Copy kube-apiserver, kube-controller-manager, kube-scheduler,
+kubelet and kubectl to a directory.
 
-On the master node, start the following daemons.
+On the master node, start the following daemons: kube-apiserver,
+kube-controller-manager, kube-scheduler. You can start first two of them in
+two modes: plain text HTTP or HTTPS. For plain text use the following
+commands:
 
+* kube-apiserver
+```
+nohup ./kube-apiserver \
+  --service-cluster-ip-range=192.168.200.0/24 \
+  --address=0.0.0.0 \
+  --etcd-servers=http://127.0.0.1:4001 \
+  --v=2 \
+  2>&1 > /dev/null &
+```
+* kube-controller-manager
+```
+nohup ./kube-controller-manager \
+  --master=127.0.0.1:8080 \
+  --v=2 \
+  2>&1 > /dev/null &
+```
+
+For HTTPS use the certificates and token should be generated beforehand.
+Please refer to [this
+document](https://coreos.com/kubernetes/docs/latest/openssl.html) on how to
+generate necessary certificates. And to [this
+document](http://kubernetes.io/docs/admin/authentication/) on how to create
+static token files.
+
+After the preparations the following commands should be run:
+* kube-apiserver
+```
+nohup ./kube-apiserver \
+  --service-cluster-ip-range=192.168.200.0/24 \
+  --address=0.0.0.0 \
+  --etcd-servers=http://127.0.0.1:4001 \
+  --v=2 \
+  --secure-port=443 \
+  --tls-cert-file=/etc/kubernetes/ssl/apiserver.pem \
+  --tls-private-key-file=/etc/kubernetes/ssl/apiserver-key.pem \
+  --client-ca-file=/etc/kubernetes/ssl/ca.pem \
+  --service-account-key-file=/etc/kubernetes/ssl/apiserver-key.pem \
+  --token-auth-file=/etc/kubernetes/auth/token.csv \
+  2>&1 > /dev/null &
 ```
-nohup ./kube-apiserver --service-cluster-ip-range=192.168.200.0/24 \
---address=0.0.0.0 --etcd-servers=http://127.0.0.1:4001 \
---v=2  2>&1 > /dev/null &
 
-nohup ./kube-controller-manager --master=127.0.0.1:8080 \
---v=2 2>&1 > /dev/null &
+* kube-controller-manager
+```
+nohup ./kube-controller-manager \
+  --master=http://127.0.0.1:8080 \
```

Comment:
This is intentional and according to documentation. Kube-controller-manager is being run only on master node and executes requests only to the kube-apiserver that resides on the same node.